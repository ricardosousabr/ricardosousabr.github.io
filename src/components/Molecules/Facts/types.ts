export interface IFacts {
  title: string;
  factsAbout: {
    id: number;
    fact: string;
  }[];
}