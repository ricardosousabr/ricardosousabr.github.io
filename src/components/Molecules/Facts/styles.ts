import styled from "styled-components";

export const Wrapper = styled.div`
  background: #282C33;
  padding: 50px 0;

`

export const FactsStyle = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  text-align: center;

`;

export const BoxText = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-bottom: 30px;
`
export const BoxFacts = styled.div`
  display: flex;
  padding: 0 50px;
  flex-wrap: wrap;
  justify-content: space-between;

`
export const BoxFact = styled.div`
  border: 1px solid #ABB2BF;
  padding: 10px;
  margin: 8px;

`

