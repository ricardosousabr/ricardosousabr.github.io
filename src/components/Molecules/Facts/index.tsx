import Text from "../../Atoms/Text";
import Container from "../../Atoms/Container";
import { IFacts } from "./types";
import { Wrapper, FactsStyle, BoxText, BoxFacts, BoxFact } from "./styles";
import Title from "../../Atoms/Title";

export default function Facts({title, factsAbout}: IFacts ) {
  return (
    <Wrapper>
      <Container>
        <FactsStyle>
          <BoxText>
            <Title >{ title }</Title>
          </BoxText>
            <BoxFacts >
            {
              factsAbout.map((fact, index) => {
                return (
                  <BoxFact key={index}>
                      <Text color="#ABB2BF" fontSize="16px" >{fact}</Text>
                  </BoxFact>
                )
              })
            }
            </BoxFacts>
        </FactsStyle>
      </Container>
    </Wrapper>
  )
}