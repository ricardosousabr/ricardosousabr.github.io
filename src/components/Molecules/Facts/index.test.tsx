import renderer from "react-test-renderer";
import Facts from "./index";
import locale from "../../../locales";

test("Link renders correctly", () => {
  const { facts } = locale['pt-br'];
  const tree = renderer.create(<Facts {...facts} />).toJSON();
  expect(tree).toMatchSnapshot();
});
