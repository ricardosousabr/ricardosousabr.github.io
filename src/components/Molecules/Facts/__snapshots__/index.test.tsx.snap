// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div
  className="sc-dlniIP jqyvKd"
>
  <div
    className="sc-gtsqUy iTcSzc"
  >
    <div
      className="sc-hKFymg iESXDq"
    >
      <div
        className="sc-eCAqax hBcfRp"
      >
        <p
          className="sc-bdnyFh iOsQJx"
          color="white"
          fontSize="20px"
        >
          Minhas curiosidades
        </p>
      </div>
      <div
        className="sc-jSFipO eAhjIz"
      >
        <div
          className="sc-gKAaef fAPlIj"
        >
          <p
            className="sc-bdnyFh hmhpbH"
            color="#ABB2BF"
            fontSize="16px"
          >
            Eu gosto mais do inverno do que do ver√£o
          </p>
        </div>
        <div
          className="sc-gKAaef fAPlIj"
        >
          <p
            className="sc-bdnyFh hmhpbH"
            color="#ABB2BF"
            fontSize="16px"
          >
            Costumo jogar com meus amigos
          </p>
        </div>
        <div
          className="sc-gKAaef fAPlIj"
        >
          <p
            className="sc-bdnyFh hmhpbH"
            color="#ABB2BF"
            fontSize="16px"
          >
            Eu gosto de pizza e batata
          </p>
        </div>
        <div
          className="sc-gKAaef fAPlIj"
        >
          <p
            className="sc-bdnyFh hmhpbH"
            color="#ABB2BF"
            fontSize="16px"
          >
            Gosto muito de viajar
          </p>
        </div>
        <div
          className="sc-gKAaef fAPlIj"
        >
          <p
            className="sc-bdnyFh hmhpbH"
            color="#ABB2BF"
            fontSize="16px"
          >
            Amo estudar
          </p>
        </div>
        <div
          className="sc-gKAaef fAPlIj"
        >
          <p
            className="sc-bdnyFh hmhpbH"
            color="#ABB2BF"
            fontSize="16px"
          >
            Apaixonado por leitura
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
